libnetwork/iptables/firewalld.go

```diff
@@ -52,11 +56,42 @@ func firewalldInit() error {
 	if connection, err = newConnection(); err != nil {
 		return fmt.Errorf("Failed to connect to D-Bus system bus: %v", err)
 	}
-	firewalldRunning = checkRunning()
+
+	tick := time.NewTicker(500 * time.Millisecond)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	firewalldRunning = false
+	cnt := 1
+
+	go func(ctx context.Context) {
+
+		select {
+		case <-tick.C:
+			firewalldRunning = checkRunning()
+			log.G(context.TODO()).Debugf("Checking status of Firewalld. Attempt # %d", cnt)
+			cnt += 1
+			if firewalldRunning {
+				return
+			}
+		case <-ctx.Done():
+			return
+		}
+	}(ctx)
+
+	time.Sleep(time.Millisecond * FireWalldInitializationTimeout)
```

Comment:
Why are you polling, but still unconditionally waiting the full timeout anyway?

Reply:
The time delay is short enough that I thought it was better to wait the full timeout period.

Reply:
Better than what? Why?

Reply:
Better than cancelling polling when firewalld is in state FAILED or RUNNING.